src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Fixer.cs

```diff
@@ -0,0 +1,199 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    public abstract class ForwardCancellationTokenToInvocationsFixer : CodeFixProvider
+    {
+        // Attempts to retrieve the invocation from the current operation.
+        protected abstract bool TryGetInvocation(
+            SemanticModel model,
+            SyntaxNode node,
+            CancellationToken ct,
+            [NotNullWhen(returnValue: true)] out IInvocationOperation? invocation);
+
+        // Looks for a ct parameter in the ancestor method or function declaration. If one is found, retrieve the name of the parameter.
+        // Returns true if a ct parameter was found and parameterName is not null or empty. Returns false otherwise.
+        protected abstract bool TryGetAncestorDeclarationCancellationTokenParameterName(
+            SyntaxNode node,
+            [NotNullWhen(returnValue: true)] out string? parameterName);
+
+        // Verifies if the specified argument was passed with an explicit name.
+        protected abstract bool IsArgumentNamed(IArgumentOperation argumentOperation);
+
+        // Retrieves the invocation expression for a conditional operation, which consists of the dot and the method name.
+        protected abstract SyntaxNode GetConditionalOperationInvocationExpression(SyntaxNode invocationNode);
+
+        public override ImmutableArray<string> FixableDiagnosticIds =>
+            ImmutableArray.Create(ForwardCancellationTokenToInvocationsAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() =>
+            WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            CancellationToken ct = context.CancellationToken;
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+
+            if (!(root.FindNode(context.Span, getInnermostNodeForTie: true) is SyntaxNode node))
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            // The analyzer created the diagnostic on the IdentifierNameSyntax, and the parent is the actual invocation
+            if (!TryGetInvocation(model, node, ct, out IInvocationOperation? invocation))
+            {
+                return;
+            }
+
+            if (!TryGetAncestorDeclarationCancellationTokenParameterName(node, out string? parameterName))
+            {
+                return;
+            }
+
+            string title = MicrosoftNetCoreAnalyzersResources.ForwardCancellationTokenToInvocationsTitle;
```

Comment:
Could you rename the resource to be `ForwardCancellationTokenToInvocationsCodeFixTitle`?

Reply:
Can you explain why? I don't see any other title string resources preceded by the keyword "Title".